pkg/scaling/scale_handler.go

```diff
@@ -624,46 +624,77 @@ func (h *scaleHandler) getScaledObjectState(ctx context.Context, scaledObject *k
 // getScaledJobMetrics returns metrics for specified metric name for a ScaledJob identified by its name and namespace.
 // It could either query the metric value directly from the scaler or from a cache, that's being stored for the scaler.
 func (h *scaleHandler) getScaledJobMetrics(ctx context.Context, scaledJob *kedav1alpha1.ScaledJob) []scaledjob.ScalerMetrics {
+	logger := log.WithValues("scaledJob.Namespace", scaledJob.Namespace, "scaledJob.Name", scaledJob.Name)
+
 	cache, err := h.GetScalersCache(ctx, scaledJob)
+	prommetrics.RecordScaledJobError(scaledJob.Namespace, scaledJob.Name, err)
 	if err != nil {
 		log.Error(err, "error getting scalers cache", "scaledJob.Namespace", scaledJob.Namespace, "scaledJob.Name", scaledJob.Name)
 		return nil
 	}
 	var scalersMetrics []scaledjob.ScalerMetrics
-	scalers, _ := cache.GetScalers()
-	for i, s := range scalers {
+	scalers, scalerConfigs := cache.GetScalers()
+	for scalerIndex, scaler := range scalers {
+		scalerName := strings.Replace(fmt.Sprintf("%T", scalers[scalerIndex]), "*scalers.", "", 1)
+		if scalerConfigs[scalerIndex].TriggerName != "" {
+			scalerName = scalerConfigs[scalerIndex].TriggerName
+		}
 		isActive := false
-		scalerType := fmt.Sprintf("%T:", s)
+		scalerType := fmt.Sprintf("%T:", scaler)
 
 		scalerLogger := log.WithValues("ScaledJob", scaledJob.Name, "Scaler", scalerType)
 
-		metricSpecs := s.GetMetricSpecForScaling(ctx)
+		metricSpecs := scaler.GetMetricSpecForScaling(ctx)
 
-		// skip scaler that doesn't return any metric specs (usually External scaler with incorrect metadata)
-		// or skip cpu/memory resource scaler
-		if len(metricSpecs) < 1 || metricSpecs[0].External == nil {
-			continue
-		}
+		for _, spec := range metricSpecs {
+			// skip scaler that doesn't return any metric specs (usually External scaler with incorrect metadata)
+			// or skip cpu/memory resource scaler
+			if len(metricSpecs) < 1 || spec.External == nil {
+				continue
+			}
 
-		metrics, isTriggerActive, _, err := cache.GetMetricsAndActivityForScaler(ctx, i, metricSpecs[0].External.Metric.Name)
-		if err != nil {
-			scalerLogger.V(1).Info("Error getting scaler metrics and activity, but continue", "error", err)
-			cache.Recorder.Event(scaledJob, corev1.EventTypeWarning, eventreason.KEDAScalerFailed, err.Error())
-			continue
-		}
-		if isTriggerActive {
-			isActive = true
-		}
+			metricName := spec.External.Metric.Name
+			metrics, isTriggerActive, latency, err := cache.GetMetricsAndActivityForScaler(ctx, scalerIndex, metricName)
+			if latency != -1 {
+				prommetrics.RecordScalerLatency(scaledJob.Namespace, scaledJob.Name, scalerName, scalerIndex, metricName, false, float64(latency))
+			}
+
+			if err != nil {
+				scalerLogger.V(1).Info("Error getting scaler metrics and activity, but continue", "error", err)
+				cache.Recorder.Event(scaledJob, corev1.EventTypeWarning, eventreason.KEDAScalerFailed, err.Error())
+				continue
+			}
+			if isTriggerActive {
+				isActive = true
```

Comment:
Why not use isTriggerActive directly?

Reply:
Good catch.
It's existing logic as scaledObject as well, so I think it's better to update with separate commit.

Reply:
`isActive` marks whether the ScaledJob is active, `isTriggerActive` is for the particular trigger in the loop